apps/vibe-coding-platform/ai/tools/create-sandbox.ts

```diff
@@ -1,75 +1,84 @@
-import type { UIMessageStreamWriter, UIMessage } from 'ai'
+import type { UIMessageChunk } from 'ai'
 import type { DataPart } from '../messages/data-parts'
 import { Sandbox } from '@vercel/sandbox'
 import { getRichError } from './get-rich-error'
 import { tool } from 'ai'
-import description from './create-sandbox.md'
+import description from './create-sandbox.description'
 import z from 'zod/v3'
+import { getWritable } from 'workflow'
 
-interface Params {
-  writer: UIMessageStreamWriter<UIMessage<never, DataPart>>
-}
+const inputSchema = z.object({
```

Comment:
The refactor for tools is very simple
1. we have to pull the execute function to the top level since `use step` only works top level. Opened https://github.com/vercel/workflow/issues/219 to track this in workflow
2. Got rid of the old writable wiring through and I'm just getting the writable directly inside steps thanks to workflow's `getWritable`

Reply:
I recommend reviewing with "hide whitespace" enabled